Library/Homebrew/cmd/update-bash.sh

```diff
@@ -0,0 +1,256 @@
+#!/bin/bash
+
+if [ -z "$HOMEBREW_BREW_FILE" ]
+then
+  echo "Error: $(basename "$0") must be called from brew!" >&2
+  exit 1
+fi
+
+brew() {
+  "$HOMEBREW_BREW_FILE" "$@"
+}
+
+which_git() {
+  local which_git
+  which_git="$(which git 2>/dev/null)"
+  if [ -n "$which_git" ] && [ "/usr/bin/git" = "$which_git" ]
+  then
+    local active_developer_dir
+    active_developer_dir="$('/usr/bin/xcode-select' -print-path 2>/dev/null)"
+    if [ -n "$active_developer_dir" ] && [ -x "$active_developer_dir/usr/bin/git" ]
+    then
+      which_git="$active_developer_dir/usr/bin/git"
+    else
+      which_git=""
+    fi
+  fi
+  echo $which_git
+}
+
+git_init_if_necessary() {
+  if [ -d ".git" ]
+  then
+    git init -q
+    git config core.autocrlf false
+    git config remote.origin.url https://github.com/Homebrew/homebrew.git
+    git config remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
+  fi
+
+  if git remote show origin -n | grep -q "mxcl/homebrew"
+  then
+    git remote set-url origin https://github.com/Homebrew/homebrew.git
+    git remote set-url --delete origin ".*mxcl\/homebrew.*"
+  fi
+}
+
+repo_var() {
+  echo "$@" |
+    sed -e "s|$HOMEBREW_PREFIX||g" \
+        -e 's|Library/Taps/||g' \
+        -e 's|[^a-z0-9]|_|g' |
+    tr "[:lower:]" "[:upper:]"
+}
+
+upstream_branch() {
+  local upstream_branch
+  upstream_branch="$(git symbolic-ref refs/remotes/origin/HEAD 2>/dev/null |
+                     sed -e 's|refs/remotes/origin/||' )"
```

Comment:
I suppose the pattern should be anchored at the beginning. Better yet, let's do more stuff directly in `bash` instead of spawning external commands all the time, especially for simple tasks like trimming a prefix from a string, which could be done like this:

```shell
upstream_branch="$(git symbolic-ref refs/remotes/origin/HEAD 2>/dev/null)"
upstream_branch="${upstream_branch#refs/remotes/origin/}"
```